pennylane/beta/tapes/tape.py

```diff
@@ -1337,3 +1660,75 @@ def jacobian(self, device, params=None, **options):
             jac[:, idx] = g.flatten()
 
         return jac
+
+    def iterator(self):
+        """Hack to make iterating over the full queue easier."""
+        queue = self.operations + self.measurements
+        for o in queue:
+            yield o
+
+
+class BatchTape(QuantumTape):
```

Comment:
Note: `QuantumTape` has a lot of methods we don't need for the `BatchTape`. It may be useful to make a `BaseQuantumTape` class that only implements the queuing behaviour, but no gradients.

Reply:
I must add, another option would be to add a flag to the ``QuantumTape`` class which is ``True`` if the tape has to be read as a batch. In a sense this is all we need here. This would also be easier for inheritance. But I it is conceptually cleaner to introduce a new class.

Reply:
The second method would, however, have the advantage that a circuit itself can be represented by a batch tape (in a setting where the entire circuits in the batch differ):